### PR TITLE
Fix auto login

### LINKED ISSUE #3772: User cannot auto login after clearing the app from ram
**Describe the bug**
Logged in user needs to login again after clearing the app from ram then re-open it.

**To Reproduce**
1. user logs into the app
2. app goes to the background
3. clear the app from ram
4. open back the app
5. ERROR: user needs to login again

**Expected behavior**
User should be auto logged in.

**Video**

https://github.com/user-attachments/assets/fe645059-0e7c-43ca-af4f-44322ba15612

**Smartphone (please complete the following information):**
Applies for all device.

**Additional context**
Original ticket is from Columbia Engage project: https://github.com/ona-health/columbia-engage-fhir-resources/issues/159

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3772

[works.webm](https://github.com/user-attachments/assets/01dce605-f603-42fc-ad8b-b3c76a757450)

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/data/remote/shared/TokenAuthenticator.kt [hunk 1]
```diff
@@ -188,6 +191,7 @@ constructor(
             account.type,
             accountManager.peekAuthToken(account, AUTH_TOKEN_TYPE),
           )
+          secureSharedPreference.deleteCredentials()
```
[the remark this answers is not in the document]
Comment (FikriMilano) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/data/remote/shared/TokenAuthenticator.kt:194]: Without this 1 line change:
When user logs out, it will immiediately logs in, because all this time, the user credentials stays even after log out where it should have been deleted.

**Comment by allan-on** [comment]
Superseded by #3851, which carries the `LoginActivity` change from this PR along with the null guard in `isCurrentRefreshTokenActive()`.

One line from here was deliberately left out: `secureSharedPreference.deleteCredentials()` inside `logout()`. Flagging it so it is not lost silently.

That entry holds `AuthCredentials(username, salt, passwordHash)`, and `findAccount()` uses the username in it to match the AccountManager account. Clearing it on logout has three effects beyond stopping the auto resume:

- Offline login stops working. `validateSavedLoginCredentials()` has nothing to compare against, so a user who logs out while online cannot log back in until they have a connection.
- The two logout paths diverge. The call only runs when the server accepts the logout, so an offline logout keeps the credentials and an online one does not.
- It quietly allows user switching. `LoginViewModel.fetchToken()` blocks a different username with `MULTI_USER_LOGIN_ATTEMPT` based on `existingCredentials != null`, so clearing them lets a second person log in on that device after an online logout. That belongs with #2779 rather than as a side effect here.

#3851 records an explicit logout marker instead, which stops the resume without touching the saved credentials.

If the user switching behaviour is actually wanted, it is worth doing deliberately: clearing credentials on both logout paths and handling the offline lockout.
